HSDir Store Events (Feature Complete)

teor2345 opened the pull request:
I have working event code, that contains all the fields @robgjansen and @ajohnson-nrl  have asked for.

From the release notes privcount/privcount@2db96dc:
* Add the PRIVCOUNT_HSDIR_CACHE_STORED event to the PrivCount Tor Patch #336
* Use tagged fields for new events #256
* Increase the maximum tor event length to 2kB #336

For the fields in the event, see privcount/privcount@be59e55.
I still have to do testing locally and on a relay.

teor2345 committed:
Add a prefix string argument to privcount_timeval_*to_epoch_str_dup()

Preparation for privcount/privcount#336.

teor2345 committed:
Add comments categorising existing event categories and existing events

Preparation for privcount/privcount#336.

teor2345 committed:
Add the EVENT_PRIVCOUNT_HSDIR_CACHE_STORED event

This event is emitted when a HS descriptor is uploaded to a v2 or v3 HSDir.
It contains the onion address (v2) or blinded public key (v3), intro point
fingerprints and count (v2 only), client auth flag, times, sizes, and caching
and validation information.

It does not contain any circuit information.

Part of privcount/privcount#336.

teor2345 committed:
Bump version: 1.0.1 → 1.1.0

robgjansen commented on src/or/rendcache.c:
```diff
@@ -696,9 +720,59 @@ rend_cache_store_v2_desc_as_dir(const char *desc)
+                                               (has_existing_cache_entry
+                                                ? "updated"
+                                                : "new"),
+                                               /* descriptor info */
```
I don't understand this coding style of having so much space to the left of separate parameters in a function block that each parameter has to be split across multiple lines. Why not just decrease the indent, so at least the result would be one parameter+comment per line? I realize that the parameters are lined up with the end of the function name and that this stile may be used elsewhere in Tor as well; but the style is very visually unappealing to me. (This is not a big deal, just my opinion.)

teor2345 replied:
I agree it's ugly, it's a compromise between make check-spaces and my editor's automatic indenting. I'll try to do better next time.

robgjansen reviewed (commented):
Looks fine to me!

teor2345 committed:
Rename PRIVCOUNT_HSDIR_CACHE_STORED to PRIVCOUNT_HSDIR_CACHE_STORE

(No D, tenses add nothing but confusion.)

sed -i.bak s/PRIVCOUNT_HSDIR_CACHE_STORED/PRIVCOUNT_HSDIR_CACHE_STORE/g \
    src/or/control.[ch]

Part of #336.

teor2345 commented:
This was squashed and merged to privcount-master in 19c6ea9cf.
